src/Microsoft.Azure.WebJobs.Host/Executors/TriggeredFunctionExecutor.cs

```diff
@@ -2,14 +2,15 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.WebJobs.Host.Protocols;
 using Microsoft.Azure.WebJobs.Host.Triggers;
 
 namespace Microsoft.Azure.WebJobs.Host.Executors
 {
-    internal class TriggeredFunctionExecutor<TTriggerValue> : ITriggeredFunctionExecutor
+    internal class TriggeredFunctionExecutor<TTriggerValue> : ITriggeredFunctionExecutor, ITriggeredFunctionExecutorWithHook
```

Comment:
I'm very open to alternatives. The criteria for me was the simplest solution that 
a) fulfilled some strict scenario requirements (it needs to be dispatched exactly at this spot in the pipeline); 
b) was relatively isolated and had a clear 'opt-in' model.  
This is a rocket-science case and I specifically didn't want it to get broad traction except from things that really needed it.  I didn't want to create a matrix of feature interactions.